shell/platform/linux/fl_platform_channel.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_platform_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_method_codec.h"
+
+struct _FlPlatformChannel {
+  FlMethodChannel parent_instance;
+};
+
+G_DEFINE_TYPE(FlPlatformChannel,
+              fl_platform_channel,
+              fl_method_channel_get_type())
+
+static void method_call_cb(FlMethodChannel* channel,
+                           const gchar* method,
+                           FlValue* args,
+                           FlMethodChannelResponseHandle* response_handle,
+                           gpointer user_data) {
+  // FlPlatformChannel* self = FL_PLATFORM_CHANNEL(user_data);
+
+  if (strcmp(method, "Clipboard.setData") == 0) {
+    g_debug("Got Clipboard.setData\n");
+    fl_method_channel_respond_not_implemented(channel, response_handle,
+                                              nullptr);
+  } else if (strcmp(method, "Clipboard.getData") == 0) {
+    g_debug("Got Clipboard.getData\n");
+    fl_method_channel_respond_not_implemented(channel, response_handle,
+                                              nullptr);
+  } else if (strcmp(method, "HapticFeedback.vibrate") == 0) {
+    g_debug("Got HapticFeedback.vibrate\n");
+    fl_method_channel_respond_not_implemented(channel, response_handle,
+                                              nullptr);
+  } else if (strcmp(method, "SystemSound.play") == 0) {
+    if (fl_value_get_type(args) != FL_VALUE_TYPE_STRING) {
+      g_warning("Ignoring unknown SystemSound.play message with unknown type");
+      fl_method_channel_respond_error(channel, response_handle, "FIXME",
+                                      nullptr, nullptr, nullptr);
+    } else {
+      const gchar* type = fl_value_get_string(args);
+      g_debug("Got SystemSound.play(\"%s\")", type);
+      // FIXME(robert-ancell): Play sound
+      fl_method_channel_respond(channel, response_handle, nullptr, nullptr);
+    }
+  } else if (strcmp(method, "SystemChrome.setPreferredOrientations") == 0) {
+    g_debug("Got SystemChrome.setPreferredOrientations\n");
+    fl_method_channel_respond_not_implemented(channel, response_handle,
+                                              nullptr);
+  } else if (strcmp(method, "SystemChrome.setApplicationSwitcherDescription") ==
```

Comment:
Probably same here (see https://github.com/flutter/flutter/issues/30712 for some discussion).